examples/gitops-kubevirt/04-upgrades/monitoring/check-upgrade-status.sh

```
@@ -0,0 +1,307 @@
+        -o jsonpath='{.data.kubeconfig}' | base64 -d > "$kubeconfig_file" 2>/dev/null || true
+
+    if [ ! -f "$kubeconfig_file" ] || [ ! -s "$kubeconfig_file" ]; then
+        log_warning "Could not extract kubeconfig for hosted cluster"
+        echo ""
+        return
+    fi
+
+    # Check cluster version in hosted cluster
+    export KUBECONFIG="$kubeconfig_file"
+
+    local cv_version=$(oc get clusterversion -o jsonpath="{.items[0].status.desired.version}" 2>/dev/null || echo "N/A")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)